### DISCUSSION TITLE
E   AttributeError: module 'webview' has no attribute 'evaluate_js'

### DISCUSSION THREAD

**Discussion by SebasCouto** [r0x0r/pywebview#1100] [Q&A] [answered]
Hello all, I am getting this error when I want to use 'evaluate_js' attribute. ¿Do we have any version where is it available?

> E   AttributeError: module 'webview' has no attribute 'evaluate_js'

I mean, I found that 'evaluate_js' attribute is an attribute that allows me to manipulate webview elements in a webview, ¿Is that right?

I leave here a screenshot. I hope this helps for a better undertanding about my concern.

<img width="195" alt="Captura de pantalla 2023-04-18 a la(s) 22 08 53" src="https://user-images.githubusercontent.com/87529848/232940617-db415957-3e3e-402a-9b71-0e0cf4b54342.png">

<img width="1047" alt="Captura de pantalla 2023-04-18 a la(s) 22 03 22" src="https://user-images.githubusercontent.com/87529848/232940571-a558ce58-6be2-4f66-8148-c6a04b5bb116.png">

Thanks in advance

**Comment by r0x0r** [comment, marked as answer]
`evaluate_js` is a function of a Window object. Change your code to 
```
window = create_window(...
window.evaluate_js(...
```

**Reply by SebasCouto** [reply]
Hi @r0x0r thanks for your reply.

I have another Q, but I am not sure where to log it, or if we can manage this from here.

¿Is there any documentation about how to map the elements from Pywebview DOM?

¿do we have any documentation for set up the rigth order of the steps to have a successful run during test?
 - For example, i share here a screenshot about elements_id objects and their steps but, I am not sure if are OK or not.

<img width="1413" alt="Captura de pantalla 2023-04-19 a la(s) 20 47 08" src="https://user-images.githubusercontent.com/87529848/233223025-ca2aff20-fe1d-4fbc-a050-8b0ef6bb7789.png">

After manually kill the python web view, I get this error:
```
test_pywebview2.py:17: in test_empretienda_web
    window.evaluate_js("document.getElementById('t_email').value = '{}';".format(email))
/Users/sebastiancouto/Library/Python/3.9/lib/python/site-packages/webview/window.py:36: in wrapper
    return function(*args, **kwargs)
/Users/sebastiancouto/Library/Python/3.9/lib/python/site-packages/webview/window.py:356: in evaluate_js
    return self.gui.evaluate_js(escaped_script, self.uid)
/Users/sebastiancouto/Library/Python/3.9/lib/python/site-packages/webview/platforms/cocoa.py:1054: in evaluate_js
    return BrowserView.instances[uid].evaluate_js(script)
E   KeyError: 'master'
```
Thanks

**Reply by r0x0r** [reply]
You can retrieve DOM elements using `window.get_elements(selector)` method.
